cmd/checkpoint/main.go

```diff
@@ -295,6 +351,11 @@ func writeCheckpointManifest(pod *v1.Pod) error {
 	return writeAndAtomicRename(path, b, 0644)
 }
 
+// isPodCheckpointer returns true if the pod is the checkpointer itself.
+func isPodCheckpointer(pod *v1.Pod) bool {
+	return strings.HasPrefix(pod.Name, "pod-checkpointer")
```

Comment:
This seems a bit fragile. Is there another way we could determine this? We can know our "self" via the downward api + inject as env vars - maybe that's an option?